tests/cases/fourslash/server/jsdocCallbackTag.ts

```diff
@@ -30,9 +30,9 @@ verify.quickInfoIs("var t: FooHandler");
 goTo.marker("2");
 verify.quickInfoIs("var t2: FooHandler2");
 goTo.marker("3");
-verify.quickInfoIs("type FooHandler2 = (eventName?: string, eventName2?: string) => any", "- What, another one?");
+verify.quickInfoIs("type FooHandler2 = (eventName?: string | undefined, eventName2?: string) => any", "- What, another one?");
```

Comment:
This seems like unwanted result of additional undefined given the parameter was written as `{string=}`

Reply:
`{string=}` can appear in not-ending positions, which still add `| undefined` to the parameter type, but _don't_ add optionality to the parameter, so the existing node walker conservatively adds `| undefined` to the type nodes any time it sees the jsdoc optional type marker, since it's nto semantically tracking if the `=` is actually implying a `?` or not. In the real world, the extra `| undefined` won't do anything if the `?` is present, so including it won't break anything, at least.